### ISSUE TITLE
Strange Makefile issue

### ISSUE THREAD

**Issue by JulioBarros** [microsoft/vscode-docker#3189]
I'm having a strange issue with the docker extension installed when I try to edit a Makefile. It used to work fine but starting a day or so ago, when I open a Makefile the syntax highlighting turns everything red and the formatting is messed up (it the tabs below the make targets are removed) when I save.

If I remove the extension the makefiles behave/edit correctly.

Could the extension be confusing Makefiles with docker files somehow?

Thanks in advance.

**Comment by rcjsuen** [comment]
@JulioBarros Please provide screenshots of the behaviour with and without the extension installed.

**Comment by rcjsuen** [comment]
@JulioBarros Also, please provide the following information:
- operating system
- version of Visual Studio Code
- version of extension installed

**Comment by JulioBarros** [comment]
@rcjsuen Thanks for the quick reply ... uninstalling and reinstalling (a couple of times) may have fixed the issue because I don't see it now.

I'll close this for now and take screenshots if I see it again.

Btw, I'm on a Mac 11.52, VSCode 1.60.0 and it reinstalled Docker plug in v1.16.1 (I don't know what it was before).

Sorry for the trouble. Thanks again.

**Comment by mcauto** [comment]
same here.

![image](https://user-images.githubusercontent.com/15080806/132179083-cafce4e2-7381-46bd-b793-deffd93b6f5a.png)

![image](https://user-images.githubusercontent.com/15080806/132179112-010f075b-d8e8-46e4-8a0c-046bebbc6103.png)

![image](https://user-images.githubusercontent.com/15080806/132179140-afefb5c6-1d18-4212-908f-caf891113c80.png)

![image](https://user-images.githubusercontent.com/15080806/132179209-8a0db99d-bfaa-4b06-ac66-34fa354ee44e.png)

**Comment by rcjsuen** [comment]
@mcauto It is strange that your screenshot shows the extension as being disabled but then there is an error.

1. Please share the contents of your Makefile.
2. Please share the output. I want to see what the `32603` error is.

**Comment by rcjsuen** [comment]
@JulioBarros @mcauto This looks like a duplicate of https://github.com/microsoft/vscode/issues/132005 which is a duplicate of https://github.com/microsoft/vscode/issues/132173.

@mcauto I am still interested in what I outlined in https://github.com/microsoft/vscode-docker/issues/3189#issuecomment-913521844 so I can reproduce the problem and see what the error is. Thank you.

**Comment by mcauto** [comment]
@rcjsuen Here!

`makefile`

```
GREEN=\n\033[1;32;40m
NC=\033[0m # No Color

PKG_LIST := $(shell go list ./...)
GO_FILES := $(shell find . -name '*.go')

# 라이브러리 설치
# tidy: 미사용 라이브러리 제거
# vendor: vendor에 라이브러리 설치
# graph: 설치된 라이브러리 그래프 확인
ref:
	@/bin/sh -c 'echo "${GREEN}[library를 vendor에 설치합니다.]${NC}"'
	@go mod tidy
	@go mod vendor -v
	@/bin/sh -c 'echo "${GREEN}[그래프 확인]${NC}"'
	@go mod graph
.PHONY: ref

# 코딩 스타일 분석
# https://github.com/golang/lint
# go get -u golang.org/x/lint/golint
lint: ref
	@/bin/sh -c 'echo "${GREEN}[정적분석(golint)을 시작합니다.]${NC}"'
	@golint -set_exit_status ${PKG_LIST}
.PHONY: lint

# 정적 분석
vet: ref
	@/bin/sh -c 'echo "${GREEN}[정적분석(vet)을 시작합니다.]${NC}"'
	@go vet ./...
.PHONY: vet

# 테스트 시작
test: ref
	@/bin/sh -c 'echo "${GREEN}[테스트를 시작합니다.]${NC}"'
	@unset LANG LC_ALL LC_MESSAGES && go test -short ${PKG_LIST}
.PHONY: test

# race condition 검사
race: ref
	@/bin/sh -c 'echo "${GREEN}[race condition을 검사합니다.]${NC}"'
	@unset LANG LC_ALL LC_MESSAGES && go test -race -v ${PKG_LIST}
.PHONY: race

# 테스트 커버리지
# go get -u github.com/axw/gocov/gocov
# go get -u github.com/matm/gocov-html
# go get -u github.com/AlekSi/gocov-xml
coverage: ref
	@/bin/sh -c 'echo "${GREEN}[test coverage를 계산합니다.]${NC}"'
	@mkdir -p .public/coverage
	@gocov test ./... | gocov-html > .public/coverage/index.html
	@gocov test ./... | gocov-xml > coverage.xml
.PHONY: coverage

# go report card
# https://github.com/gojp/goreportcard
report: ref
	@/bin/sh -c 'echo "${GREEN}[go report card를 생성합니다]${NC}"'
	@goreportcard-cli -d src
.PHONY: report

# 빌드
build: ref
	@/bin/sh -c 'echo "${GREEN}[빌드를 시작합니다]${NC}"'
	@mkdir -p bin
	@go build -o bin/
	@ls -al bin
.PHONY: build

VERSION := $(shell cat main.go | grep "const version" | cut -d ' ' -f4 | cut -d '"' -f2)
REGISTRY=
build-docker-base: ref 
	@/bin/sh -c 'echo "${GREEN}[base image 빌드를 시작합니다]${NC}"'
	@docker build \
		-f "base.Dockerfile" \
		-t ${REGISTRY}/base:latest .
	@docker build \
		-f "base.Dockerfile" \
		-t ${REGISTRY}/base:v1.0.1 .
.PHONY: build-docker

build-docker: build-docker-base
	@/bin/sh -c 'echo "${GREEN}[image 빌드를 시작합니다]${NC}"'
	@docker build \
		-f "base.Dockerfile" \
		-t ${REGISTRY}/app:latest .
	@docker build \
		-f "base.Dockerfile" \
		-t ${REGISTRY}/app:v${VERSION} .
.PHONY: build-docker-base

install: build
	@/bin/sh -c 'echo "${GREEN}[설치를 시작합니다]${NC}"'
	@mv bin/* ${GOPATH}/bin/
.PHONY:

clean:
	@rm -rf bin vendor
.PHONY: clean
```

`vscode output (Dockerfile Language Server)`

```
[Error - 1:15:23 PM] Request textDocument/semanticTokens/full failed.
  Message: Request textDocument/semanticTokens/full failed with message: Maximum call stack size exceeded
  Code: -32603 
[Error - 1:15:24 PM] Request textDocument/semanticTokens/full failed.
  Message: Request textDocument/semanticTokens/full failed with message: Maximum call stack size exceeded
  Code: -32603 
```

**Comment by rcjsuen** [comment]
@mcauto Thank you for sharing that! I can reproduce the problem on my end. 👍

**Comment by bwateratmsft** [comment]
I'm going to reopen this for now. It's supposed to be fixed in VSCode 1.60.1, we'll keep an eye on it.

As a workaround according to [this comment](https://github.com/microsoft/vscode/issues/132005#issuecomment-912840712), you can disable the AI language detection with this setting in `settings.json`:

```json
"[makefile]": {
        "workbench.editor.languageDetection": false
},
```

**Comment by bwateratmsft** [comment]
I verified the fix they described in https://github.com/microsoft/vscode/issues/132005. Please note, it requires the user to override the language mode manually. The AI classification system is still (incorrectly) classifying it as a Dockerfile, but this is outside of the extension's control.
